File: mobile/lib/api/categories_api.dart
import 'api_client.dart';

Future<dynamic> listCategories(String token) {
  return apiRequest('/categories', token: token);
}

Future<dynamic> createCategory(String token, Map<String, dynamic> category) {
  return apiRequest('/categories', method: 'POST', body: category, token: token);
}

Future<dynamic> updateCategory(String token, int id, Map<String, dynamic> category) {
  return apiRequest('/categories/$id', method: 'PUT', body: category, token: token);
}

Future<dynamic> deleteCategory(String token, int id) {
  return apiRequest('/categories/$id', method: 'DELETE', token: token);
}

File: web/src/features/admin/categories/categoriesApi.ts
import { apiClient } from '@/services/apiClient';
import type { ServiceCategory, ServiceCategoryInput } from './types';

interface ApiEnvelope<T> {
  success: boolean;
  data: T;
}

export async function fetchCategories(): Promise<ServiceCategory[]> {
  const { data } = await apiClient.get<ApiEnvelope<ServiceCategory[]>>('/categories');
  return data.data;
}

export async function createCategory(input: ServiceCategoryInput): Promise<ServiceCategory> {
  const { data } = await apiClient.post<ApiEnvelope<ServiceCategory>>('/categories', input);
  return data.data;
}

export async function updateCategory(
  id: number,
  input: ServiceCategoryInput,
): Promise<ServiceCategory> {
  const { data } = await apiClient.put<ApiEnvelope<ServiceCategory>>(`/categories/${id}`, input);
  return data.data;
}

export async function deleteCategory(id: number): Promise<void> {
  await apiClient.delete(`/categories/${id}`);
}
